### ISSUE TITLE
Handle 404 error while adding a pet to an owner

### ISSUE THREAD

**Issue by arey** [spring-petclinic/spring-petclinic-rest#130]
The `POST /owners/{ownerId}/pets` does not return a 404 error when the `ownerId` does not exist but an error `400`
```
{ 
   "className": "org.springframework.dao.DataIntegrityViolationException", 
   "exMessage": "could not execute statement [integrity constraint violation: foreign key no parent ; FK_PETS_OWNERS table: PETS value: 222] [insert into pets (birth_date,name,owner_id,type_id,id) values (?,?,?,?,default)]; SQL [insert into pets (birth_date,name,owner_id,type_id,id) values (?,?,?,?,default)]; constraint [FK_PETS_OWNERS]"
}
```

Same behaviour when a Pet does not exist

**Comment by earlspilner** [comment]
Hello,

I recently made updates to exception handling mechanism, specifically focusing on the `POST /owners/{ownerId}/pets` endpoint to correctly return 404 errors for non-existing owners and pets, and refactored error handling to use Java `record` for error information.

After these changes, I noticed that several tests are now failing:

- **`testAddPetError`**:
  - Expected status: 400 Bad Request
  - Actual status: 500 Internal Server Error

- **`testCreateUserError`**:
  - Expected status: 400 Bad Request
  - Actual status: 500 Internal Server Error

The test failures seem to be related to the status codes returned by the updated error handling logic. 

**Questions:**

1. **Expected Behavior**: Should these errors be returning a `500 Internal Server Error` instead of `400 Bad Request` now? If so, should I update tests to expect a `500` status code?
2. **Error Handling**: Are there specific types of exceptions or error scenarios where it should still return a `400 Bad Request` instead of `500 Internal Server Error`? If so, how should I handle these cases in updated error handling logic?

Any guidance on aligning error handling with the expected behavior and updating the tests accordingly would be greatly appreciated.

Thank you!
